File: frontend/src/services/trades.ts
import { api } from "./api"
import type { Trade, TradeCreate, ImportBatch } from "../types/api"

export const tradesService = {
  list: () => api.get<Trade[]>("/api/v1/trades"),
  create: (payload: TradeCreate) => api.post<Trade>("/api/v1/trades", payload),
  get: (id: string) => api.get<Trade>(`/api/v1/trades/${id}`),
  uploadCsv: (file: File) => {
    const formData = new FormData()
    formData.append("file", file)
    return api.postForm<ImportBatch>("/api/v1/imports", formData)
  },
  listImports: () => api.get<ImportBatch[]>("/api/v1/imports"),
}

File: backend/app/api/v1/trades.py
import uuid

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.deps import get_current_user
from app.models.trade import Trade
from app.models.user import User
from app.schemas.trade import TradeCreate, TradeResponse

router = APIRouter(prefix="/api/v1/trades", tags=["trades"])


@router.get("", response_model=list[TradeResponse])
def list_trades(
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return db.query(Trade).filter(Trade.user_id == current_user.id).order_by(Trade.entry_time.desc()).all()


@router.post("", response_model=TradeResponse, status_code=status.HTTP_201_CREATED)
def create_trade(
    payload: TradeCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    trade = Trade(**payload.model_dump(), user_id=current_user.id)
    db.add(trade)
    db.commit()
    db.refresh(trade)
    return trade


@router.get("/{trade_id}", response_model=TradeResponse)
def get_trade(
    trade_id: uuid.UUID,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    trade = db.query(Trade).filter(Trade.id == trade_id, Trade.user_id == current_user.id).first()
    if not trade:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Trade not found")
    return trade